src/nwb2bids/bids_models/_dataset_description.py

```diff
@@ -7,6 +8,31 @@
 from ._base_metadata_model import BaseMetadataModel
 
 
+class GeneratedByNwb2bids(BaseMetadataModel):
```

Comment:
Should probably split this into a base 'GenertatedBy' model to use in proper typing below

Reply:
Then the child class of that, specific to the nwb2bids generation, can set defaults, etc.